benchmarks/triangulated_surface_distance/generator.hpp

```diff
@@ -0,0 +1,269 @@
+/****************************************************************************
+ * Copyright (c) 2024 by the ArborX authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX_HyperTriangle.hpp>
+
+#include <Kokkos_MathematicalConstants.hpp>
+
+#if KOKKOS_VERSION >= 40400
+#include <Kokkos_Array.hpp>
+using Array = Kokkos::Array;
+#else
+template <class T, size_t N>
+struct Array
+{
+  static constexpr auto size() { return N; }
+  KOKKOS_FUNCTION constexpr T &operator[](int i) { return _data[i]; }
+  KOKKOS_FUNCTION constexpr T const &operator[](int i) const
+  {
+    return _data[i];
+  }
+
+  T _data[N];
+};
+#endif
+
+static auto icosahedron()
+{
+  auto a = Kokkos::numbers::phi_v<float>;
+  auto b = 1.f;
+
+  using Point = ArborX::ExperimentalHyperGeometry::Point<3>;
+
+  std::vector<Point> vertices;
+
+  vertices.push_back(Point{0, b, -a});
+  vertices.push_back(Point{b, a, 0});
+  vertices.push_back(Point{-b, a, 0});
+  vertices.push_back(Point{0, b, a});
+  vertices.push_back(Point{0, -b, a});
+  vertices.push_back(Point{-a, 0, b});
+  vertices.push_back(Point{0, -b, -a});
+  vertices.push_back(Point{a, 0, -b});
+  vertices.push_back(Point{a, 0, b});
+  vertices.push_back(Point{-a, 0, -b});
+  vertices.push_back(Point{b, -a, 0});
+  vertices.push_back(Point{-b, -a, 0});
+
+  std::vector<Array<int, 3>> triangles;
+
+  triangles.push_back({2, 1, 0});
+  triangles.push_back({1, 2, 3});
+  triangles.push_back({5, 4, 3});
+  triangles.push_back({4, 8, 3});
+  triangles.push_back({7, 6, 0});
+  triangles.push_back({6, 9, 0});
+  triangles.push_back({11, 10, 4});
+  triangles.push_back({10, 11, 6});
+  triangles.push_back({9, 5, 2});
+  triangles.push_back({5, 9, 11});
+  triangles.push_back({8, 7, 1});
+  triangles.push_back({7, 8, 10});
+  triangles.push_back({2, 5, 3});
+  triangles.push_back({8, 1, 3});
+  triangles.push_back({9, 2, 0});
+  triangles.push_back({1, 7, 0});
+  triangles.push_back({11, 9, 6});
+  triangles.push_back({7, 10, 6});
+  triangles.push_back({5, 11, 4});
+  triangles.push_back({10, 8, 4});
+
+  return std::make_tuple(vertices, triangles);
+}
+
+void convertTriangles2EdgeForm(std::vector<Array<int, 3>> &triangles,
```

Comment:
This is not declared static so I suppose you did not need to specify internal linkage above.